tap_facebook/__init__.py

```diff
@@ -486,7 +486,9 @@ def sync_batches(self, stream_objects):
 
         # Ensure the final batch is executed
         api_batch.execute()
-        return str(pendulum.parse(latest_lead[self.replication_key]))
+        # Return lead with maximum replication_key for bookmark if any leads found
+        if latest_lead:
+            return str(pendulum.parse(latest_lead[self.replication_key]))
```

Comment:
Added above condition because the tap was breaking(parsing None as a date error) when no data is available for leads.
The above function returns the maximum replication key's value out of all leads for writing bookmark.